components/beehiiv/app/beehiiv.app.ts

```diff
@@ -1,13 +1,100 @@
 import { defineApp } from "@pipedream/types";
+import { axios } from "@pipedream/platform";
 
 export default defineApp({
   type: "app",
   app: "beehiiv",
-  propDefinitions: {},
+  propDefinitions: {
+    email: {
+      type: "string",
+      label: "Email",
+      description: "The email address of the subscriber.",
+    },
+    publicationId: {
+      type: "string",
+      label: "Publication Id",
+      description: "The ID of the publication you want to subscribe to.",
+      async options() {
+        return this.getPublicationOpts();
+      },
+    },
+    reactivateExisting: {
+      type: "boolean",
+      label: "Reactivate Existing",
+      description: "Whether or not to reactivate the subscriber if they have already unsubscribed. **This option should be used only if the subscriber is knowingly resubscribing**. default: `false`.",
+      optional: true,
+    },
+    sendWelcomeEmail: {
+      type: "boolean",
+      label: "Send Welcome Email",
+      description: "Whether or not to send a welcome email to the subscriber. default: `false`.",
+      optional: true,
+    },
+    utmSource: {
+      type: "string",
+      label: "UTM Source",
+      description: "The source of the subscriber.",
+      optional: true,
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _getUrl(path) {
+      return `https://api.beehiiv.com/v1${path}`;
+    },
+    _getHeaders(headers = {}) {
+      return {
+        "X-ApiKey": this.$auth.api_key,
+        "Content-Type": "application/json",
+        ...headers,
+      };
+    },
+    _getRequestParams(opts: any) {
+      return {
+        ...opts,
+        url: this._getUrl(opts.path),
+        headers: this._getHeaders(),
+      };
+    },
+    filterEmptyValues(obj) {
```

Comment:
This method shouldn't be needed as `axios` already clears empty values. [See here](https://github.com/PipedreamHQ/pipedream/blob/66be3412a0aff553b60efbff5912237904eaf2c3/platform/lib/axios.ts#L53)

Reply:
it seems that it only removes "undefined" props in the object, but it seems to work on this scenario